### DISCUSSION TITLE
Help a beginner to add existing files and a folder to repository

### DISCUSSION THREAD

**Discussion by Nummuliitti** [community/community#109355] [New to GitHub] [answered]
Hello

I have a folder tree "D:/test/src" 

In this "/src" folder i have 2 files "App.jsx" and "main.jsx". They must be in this folder, in order for the application to work.

How can i create a repository to monitor these two files, and the whole "/src" folder?

I use github desktop. If i try to select the /src folder, it creates a subfolder /src/src and that is not what is needed.

Thank you

**Comment by NipulM** [comment]
Hello! 👋🏼

In the menu bar, navigate to File > Add Local Repository. Once you've selected the parent folder (D:/test), ensure to check the box next to "Include untracked files." Then, click "Choose."

To ignore unnecessary files:
After allowing GitHub some time to scan the folder, click the "Filter Files" button in the top right corner. Select "Show Ignored Files," and you may spot unnecessary files that you don't want to track. Simply right-click on them and add them to .gitignore.

Once you've provided a commit message, commit and push to the main branch. ;)

Hope this helps!

**Comment by Nummuliitti** [comment]
Hello and thank you

When i go to File > Add local repository and select D:/test folder, the program does not show a checkbox "include untracked files". Instead it immediatily says "This directory does not appear to be a Git repository. Would you like to "create a repository" here instead"?

I am unable to do anything else than to press "create a repository" link. When i do that, a window opens and all i can do is a create a new repository. In the opening window if i enter "Name: src" and "Local path: D:/test/src" - the program will create a subfolder "D:/test/src/src" and it is wrong again.

What to do?

**Reply by NipulM** [reply, marked as answer]
Ohh I see,

- Open the GitHub web interface and click on the "+" button to create a new repository.
- Name your repository (e.g., "test") and provide an optional description.
- Choose whether to make the repository private or public.
- Do not add a README file at this stage.
- Click on "Create repository."
- Once created, you'll see a link like this: https://github.com/<YOUR_USERNAME>/test.git. Copy this link.

Now, in your terminal (command prompt):

Navigate to the directory where you want to initialise your local repository. For example:
`cd D:/test/src`

Initialize a Git repository in this directory:
`git init`

Stage your initial files and commit them:
```
git add .
git commit -m "Initial commit"
```

Link your local repository to the GitHub repository using the copied link:
```
git remote add origin https://github.com/<YOUR_USERNAME>/test.git 
git branch -M main
git push -u origin main
```

Make sure to replace <YOUR_USERNAME> with your actual GitHub username and ensure that you have the necessary permissions to push to the repository. 
Also, if you're doing this for the first time on a new machine, you may need to configure your Git user details using git config --global user.name "Your Name" and git config --global user.email "your.email@example.com".

**Reply by Nummuliitti** [reply]
Thank you, i was able to do this now.

Thank you also for yash2026 for additional answer.

**Reply by m4rkbello** [reply]
> Ohh I see,
> 
> * Open the GitHub web interface and click on the "+" button to create a new repository.
> * Name your repository (e.g., "test") and provide an optional description.
> * Choose whether to make the repository private or public.
> * Do not add a README file at this stage.
> * Click on "Create repository."
> * Once created, you'll see a link like this: https://github.com/<YOUR_USERNAME>/test.git. Copy this link.
> 
> Now, in your terminal (command prompt):
> 
> Navigate to the directory where you want to initialise your local repository. For example: `cd D:/test/src`
> 
> Initialize a Git repository in this directory: `git init`
> 
> Stage your initial files and commit them:
> 
> ```
> git add .
> git commit -m "Initial commit"
> ```
> 
> Link your local repository to the GitHub repository using the copied link:
> 
> ```
> git remote add origin https://github.com/<YOUR_USERNAME>/test.git 
> git branch -M main
> git push -u origin main
> ```
> 
> Make sure to replace <YOUR_USERNAME> with your actual GitHub username and ensure that you have the necessary permissions to push to the repository. Also, if you're doing this for the first time on a new machine, you may need to configure your Git user details using git config --global user.name "Your Name" and git config --global user.email "[your.email@example.com](mailto:your.email@example.com)".

ohahay

**Comment by its21yash** [comment]
To create a repository to monitor the 'App.jsx' and 'main.jsx' files along with the entire '/src' folder using GitHub Desktop, you can follow these steps:

1. Create a New Repository:
- Open GitHub Desktop.
- Click on the "File" menu and select "New Repository".
- Choose a name and location for your repository.
- Click on "Create Repository".

2. Copy Files into Repository:
- Copy the 'App.jsx' and 'main.jsx' files into the repository folder you created in step 1.

3. Initialize Repository:
- Open GitHub Desktop and select the repository you just created.
- Click on the "Repository" menu and select "Repository Settings".
- Check the box that says "Initialize this repository with a README" if you want to create a README file.
- Click on "Initialize".

Commit Changes:
- You should now see the 'App.jsx' and 'main.jsx' files listed under "Changes" in GitHub Desktop.
- Enter a commit message and click on "Commit to main".

Publish Repository:
- Click on the "Repository" menu and select "Publish repository".
- Choose a name for your repository on GitHub and click on "Publish repository".


Now, your repository should be set up to monitor the 'App.jsx' and 'main.jsx' files along with the entire '/src' folder. You can continue to make changes to these files locally and commit and push them to the repository using GitHub Desktop.

**Reply by m4rkbello** [reply]
it's not working to me.
